### ISSUE TITLE
Sitemap generated by cron have wrong image URL (cache path)

### ISSUE THREAD

**Issue by dimaMackert** [magento/magento2#27458]
### Preconditions (*)

1. Magento 2.4
2. Create 2 stores
3. Add at least one product to each of those store
4. Add at least one image to each of the products
5. Create Sitemap for one of the stores.

### Steps to reproduce (*)

1. Generate a sitemap via backend
2. Generate a sitemap via cron
3. You will see, there are two different cache paths

### Expected result (*)

I found almost the same issue here: https://github.com/magento/magento2/pull/24605, here you generate the sitemap via backend. 
In my case i let generate the sitemap with cron schedule. The problem is i get a wrong  cache path

Here is a example wit the wrong path. 
```
/media/catalog/product/cache/cb8ca2a9db68e4f593ce69a9ee07e2b0/p/r/test.png
```

### Actual result (*)

After next changes I generate another sitemap I get the correct cache path 
```

/media/catalog/product/cache/9b58a33b64c72152de346a8490e93157/p/r/test.png
```
![Peek 2020-08-28 14-45](https://user-images.githubusercontent.com/51681379/91557515-44840100-e93d-11ea-8c65-0cd18987de5b.gif)

**Additional information** 
I found out, that the cron generates the sitemap with:
\Magento\Sitemap\Model\Observer:scheduledGenerateSitemaps()

The scheduledGenerateSitemaps() method does not implement the appEmulation->startEnvironmentEmulation() like it was fixed here: https://github.com/magento/magento2/pull/24605
I changed the method and it worked for me

```
/**
     * Generate sitemaps
     *
     * @return void
     * @throws \Exception
     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
     */
    public function scheduledGenerateSitemaps()
    {
        $errors = [];
        $recipient = $this->scopeConfig->getValue(
            Observer::XML_PATH_ERROR_RECIPIENT,
            ScopeInterface::SCOPE_STORE
        );
        // check if scheduled generation enabled
        if (!$this->scopeConfig->isSetFlag(
            self::XML_PATH_GENERATION_ENABLED,
            ScopeInterface::SCOPE_STORE
        )
        ) {
            return;
        }

        $collection = $this->collectionFactory->create();
        /* @var $collection \Magento\Sitemap\Model\ResourceModel\Sitemap\Collection */
        foreach ($collection as $sitemap) {
            /* @var $sitemap \Magento\Sitemap\Model\Sitemap */
            try {
                $this->appEmulation->startEnvironmentEmulation(
                    $sitemap->getStoreId(),
                    Area::AREA_FRONTEND,
                    true
                );
                $sitemap->generateXml();
                $this->appEmulation->stopEnvironmentEmulation();
            } catch (\Exception $e) {
                $errors[] = $e->getMessage();
            }
        }
        if ($errors && $recipient) {
            $this->emailNotification->sendErrors($errors);
        }
    }
```
Other possible solution: https://github.com/magento/magento2/pull/27476

**Comment by engcom-Delta** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Delta. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Delta, here is your [Magento instance](https://i-27458-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27458-2-4-develop.instances.magento-community.engineering/admin_2d51](https://i-27458-2-4-develop.instances.magento-community.engineering/admin_2d51)
Login: `d6358fbd` Password: `d05b6e685384`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37200` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by thiaramus** [comment]
The internal Magento team is working on this issue right now.
The fix will be based on commits from the author and delivered according to our internal processes. All original commits made by the author will be preserved.

**Comment by sdzhepa** [comment]
Changed priority and severity according to the internal jira ticket
https://jira.corp.magento.com/browse/MC-37306

**Comment by zaimg05** [comment]
tengo el mismo problema  trabajo con magento 2.3.3 y cuento con mas 5 de tiendas  pero las imagenes no se cargan en las demas tiendas solo en la principal 
las rutas de la url son diferente
¿Como puedo solucionarlo?

**Comment by sdzhepa** [comment]
Hello @dimaMackert 

This issue has been fixed by Magento team in the scope of jira ticket `MC-37306`
The PR with fix was delivered into `2.4-develop` branch.

Here you can find commits https://github.com/magento/magento2/search?q=MC-37306&type=commits
